servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Single.java

```diff
@@ -535,6 +536,30 @@ public final Completable flatMapCompletable(Function<? super T, ? extends Comple
         return new SingleConcatWithPublisher<>(this, next, executor);
     }
 
+    /**
+     * Create a new {@link Single} that emits the results of a specified combinator {@link Function} to items emitted
+     * by {@code singles}.
+     * <p>
+     * From a sequential programming point of view this method is roughly equivalent to the following:
+     * <pre>{@code
+     *      CompletableFuture<T> f1 = ...; // this
+     *      CompletableFuture<T2> other = ...;
+     *      CompletableFuture.allOf(f1, other).get(); // wait for all futures to complete
+     *      return zipper.apply(f1.get(), other.get());
+     * }</pre>
+     * @param other The other {@link Single} to zip with.
+     * @param zipper Used to combine the completed results for each item from {@code singles}.
+     * @param <T2> The type of {@code other}.
+     * @param <R> The result type of the combinator.
```

Comment:
Should we name the argument combinator to keep it consistent?

Reply:
I'll rename to `zipper` for consistency.